test/test_server.py

```diff
@@ -76,6 +77,7 @@ def test_api_search_for_no_response(mock_bad_request):
             mock_bad_request.assert_called_with([404, 'No response',
                                                  'google:fossasia'])
             assert resp == "Mock Response"
+'''
```

Comment:
@realslimshanky can you please see this test I skipped(commen out)...
`test_api_searh_for_no_response`, why line 77 will ever be called. My point is why `bad_request` function will be called when `'/api/v1/search/google?query=fossasia'` fetches correct results, so it will not lead to calling up bad request.

Reply:
Actually, it is not testing for response from the server, it's testing the response already stored in catch. And since the request is made only once it's not present in cache.

Reply:
So if not present in the cache, which code ideally should be executed, `bad_request` should have been called ?
Which statements are we talking about here ... Please help.

Reply:
I have not worked with pytest/mock yet and would take some time to fully understand in order to assist you better. I would like to request contributors who have worked on this module to please help out.
cc: @vaibhavsingh97  @bhaveshAn

Reply:
ping @cclauss @SaptakS

Reply:
@cclauss @bhaveshAn @vaibhavsingh97 please  suggest.